### PR TITLE
Move `ExternalDeploymentTask` to core

### PR DESCRIPTION
**Pull request by kaxil**
Related to https://github.com/astronomer/astronomer-providers/pull/683#discussion_r983774615

This should be in the external_task.py

### CODE REVIEW

**Comment by bharanidharan14** [comment]
Example Dag also need to be moved to core example dag folder

File: tests/core/triggers/test_external_task.py [hunk 1]
```diff
@@ -133,3 +140,99 @@ def test_serialization(self):
...
+        "endpoint": "test-endpoint",
+        "extra_options": {},
+        "headers": {"Content-Type": "application/json"},
+        "http_conn_id": "http_default",
+        "poke_interval": 5.0,
+    }
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.http.hooks.http.HttpHookAsync.run")
```
Comment (bharanidharan14) [inline tests/core/triggers/test_external_task.py:168]: All mock paths should be changed pointing to core

**Review by bharanidharan14 — CHANGES_REQUESTED** [review]
Taking this branch and making changes to  fix the Example DAG, test case import path and serialisation class in trigger

**Comment by rajaths010494** [comment]
Looks good to me.

**Review by rajaths010494 — APPROVED** [review]

**Review by bharanidharan14 — APPROVED** [review]

**Review by pankajastro — APPROVED** [review]

**Pull request merged**